### PR TITLE
Website: Add automatic citation data

### PR DESCRIPTION
**Pull request by Pandapip1**
Depends on https://github.com/ethereum/EIPs/pull/6514

### CODE REVIEW

File: _includes/head.html [hunk 1]
```diff
@@ -13,6 +13,31 @@
...
+      {%- else -%}
+        <meta name="citation_author" content="{{ author }}">
+      {%- endif -%}
+    {% endfor %}
+    {% if page.status == "Final" %}
+      {% assign pubdate = page.finalized | split: '-' %}
+      <meta name="citation_publication_date" content="{% for part in pubdate %}{{ part }}{% if forloop.last == false %}/{% endif %}/{% endfor %}">
+    {% endif %}
+    {% assign createdate = page.created | split: '-' %}
+    <meta name="citation_online_date" content="{% for part in createdate %}{{ part }}{% if forloop.last == false %}/{% endif %}/{% endfor %}">
```
Comment (SamWilsn) [inline _includes/head.html:38]: Would something like this work?

```suggestion
    <meta name="citation_online_date" content="{{ createdate | join: '/' }}">
```

File: _includes/head.html [hunk 2]
```diff
@@ -13,6 +13,31 @@
...
+      {%- elsif author contains "(@" -%}
+        {%- assign authorparts=author|split:"(@" -%}
+        <meta name="citation_author" content="{{ authorparts[0]|strip }}">
+      {%- else -%}
+        <meta name="citation_author" content="{{ author }}">
+      {%- endif -%}
+    {% endfor %}
+    {% if page.status == "Final" %}
+      {% assign pubdate = page.finalized | split: '-' %}
+      <meta name="citation_publication_date" content="{% for part in pubdate %}{{ part }}{% if forloop.last == false %}/{% endif %}/{% endfor %}">
```
Comment (SamWilsn) [inline _includes/head.html:35]: Maybe:

```suggestion
      <meta name="citation_publication_date" content="{{ pubdate | join: '/' }}">
```

File: _includes/head.html [hunk 3]
```diff
@@ -13,6 +13,31 @@
     <meta name="description" content="{{page.description | escape }}" />
     <meta property="og:description" content="{{page.description | escape }}" />
     <meta name="twitter:description" content="{{page.description | escape }}" />
+    {% if page.category == "ERC" %}
```
Comment (SamWilsn) [inline _includes/head.html:16]: I kinda feel like we're at the point where we should do something like:

```liquid
{% if page.category == "ERC" %}
    {% assign prefix = "ERC" %}
{% else %}
    {% assign prefix = "EIP" %}
{% endif %}
```

File: _includes/head.html [hunk 4]
```diff
@@ -13,6 +13,31 @@
...
+      <meta name="citation_title" content="ERC-{{ page.eip }}: {{ page.title | escape }}">
+    {% else %}
+      <meta name="citation_title" content="EIP-{{ page.eip }}: {{ page.title | escape }}">
+    {% endif %}
+    {%- assign authors=include.authors|split:"," -%}
+    {% for author in authors %}
+      {%- if author contains "<" -%}
+        {%- assign authorparts=author|split:"<" -%}
+        <meta name="citation_author" content="{{ authorparts[0]|strip }}">
```
Comment (SamWilsn) [inline _includes/head.html:25]: Do all of these need `| escape`?

File: _includes/head.html (hunk 1 above)
Comment (Pandapip1) [reply _includes/head.html:38]: If you're sure this works, okay.

File: _includes/head.html (hunk 4 above)
Comment (Pandapip1) [reply _includes/head.html:25]: Yes.

File: _includes/head.html (hunk 3 above)
Comment (Pandapip1) [reply _includes/head.html:16]: Yea, but this isn't the place to do that.

**Comment by Pandapip1** [comment]
Superseded by rewrite

**Pull request closed without merging**